src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -1,26 +1,89 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace System {
-    public struct Range
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System
+{
+
+    public struct Range : IEnumerable<int>
     {
-        public Range(long index, long length)
+        public const int UnboundedFirst = Int32.MinValue;
+        public const int UnboundedLast = Int32.MaxValue;
+        public const uint UnboundedLength = UInt32.MaxValue;
+
+        public readonly int First;
+        public readonly int Last; // Last is exclusive
```

Comment:
I don't have a strong opinion here, but we should follow-up on naming: First+Last vs. Start+End.

Reply:
Last can't be exclusive, but End means that everywhere.

Reply:
For example, the `System.Linq.Enumerable` class has the `First<T>` and `Last<T>` methods. The second returns the last element in the sequence or throws. So having the `Last` field which holds an excluded value would be confusing.

```csharp
var range = new Range(0, 10);
var last1 = range.Last; // using the Last field, 10
var last2 = range.Last(); // using the Enumerable.Last<T> method, 9
```
But they should return the same result. Isn't?

Reply:
Good point. I will change it.